elements/map/src/layer.ts

```diff
@@ -7,16 +9,38 @@ import { EOxMap } from "../main";
  * @returns Layer
  */
 export function getLayerById(EOxMap: EOxMap, layerId: string) {
-  // get mapbox-style layer or manually generated ol layer
-  const layers = EOxMap.map.getLayers().getArray();
+  const flatLayers = getFlatLayersArray(EOxMap.map.getLayers().getArray());
+  // get mapbox-style layer or manually generated ol layer, both group or regular layers
+
   const layer =
-    layers.find((l) => l.get("id") === layerId) ||
-    layers
+    flatLayers.find((l) => l.get("id") === layerId) ||
```

Comment:
Maybe this could be overrideable (e.g. as third, optional, function parameter), because there could be some edge cases where it's not "id"?